File: src/main/java/edu/gbcg/dbInteraction/dbSelector/reddit/comments/RedditComSelector.java
/*
 * Copyright (c) 2018 Sean Grimes. All rights reserved.
 * License: MIT License
 */

package edu.gbcg.dbInteraction.dbSelector.reddit.comments;

import edu.gbcg.configs.Finals;
import edu.gbcg.dbInteraction.dbSelector.RSMapper;
import edu.gbcg.dbInteraction.dbSelector.SelectionWorker;
import edu.gbcg.dbInteraction.dbSelector.Selector;
import edu.gbcg.configs.DBLocator;

import java.util.ArrayList;
import java.util.List;

public class RedditComSelector extends Selector {

    public RedditComSelector(){
        this.tableName = Finals.COM_TABLE_NAME;
    }

    /**
     * Execute a SQL query across all DB shards in parallel
     * @param SQLStatement
     * @return A list of RSMapper objects containing any results
     */
    public List<RSMapper> generalSelection(String SQLStatement){
        List<String> DBs = DBLocator.redditComsAbsolutePaths();
        verifyDBsExist(DBs);

        List<SelectionWorker> workers = new ArrayList<>();
        for(int i = 0; i < DBs.size(); ++i)
            workers.add(new SelectionWorker(DBs.get(i), SQLStatement, new CommentSetMapper()));
        return genericSelect(workers, SQLStatement);
    }
}

File: src/main/kotlin/edu/antevortadb/dbInteraction/dbSelector/reddit/comments/RedditComSelector.kt
/*
 * Copyright (c) 2018 Sean Grimes. All Rights Reserved.
 * License: MIT
 */

package edu.antevortadb.dbInteraction.dbSelector.reddit.comments

import edu.antevortadb.configs.DBLocator
import edu.antevortadb.configs.Finals
import edu.antevortadb.dbInteraction.columnsAndKeys.RedditComs
import edu.antevortadb.dbInteraction.dbSelector.RSMapper
import edu.antevortadb.dbInteraction.dbSelector.SelectionWorker
import edu.antevortadb.dbInteraction.dbSelector.Selector

class RedditComSelector : Selector() {
    init {
        this.tableName = Finals.REDDIT_COM_TABLE
        this.listOfColumns = RedditComs.columnNames()
    }

    override fun generalSelection(SQLStatement: String): List<RSMapper> {
        val dbs = DBLocator.redditComsAbsolutePaths()
        verifyDBsExist(dbs)

        val workers = ArrayList<SelectionWorker>()
        for(i in 0 until dbs.size)
            workers.add(SelectionWorker(dbs[i], SQLStatement, CommentSetMapper()))
        return genericSelect(workers, SQLStatement)
    }
}
